html/modules/custom/reliefweb_api/reliefweb_api.module

```diff
@@ -145,6 +163,70 @@ function reliefweb_api_execute(array $options, $entity_type_id) {
   }
 }
 
+/**
+ * Check if taxonomy terms require re-indexing of tagged content.
+ *
+ * Store terms for re-indexing so we can process them later in batches.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $entity
+ *   Entity that might require re-indexing.
+ * @param string $bundle
+ *   Taxonomy type of entity.
+ */
+function reliefweb_api_check_reindexing(EntityInterface $entity, $bundle) {
+  $needs_reindex = FALSE;
+  // Get unchanged entity for comparison.
+  // Tried loadUnchanged() via hooks _entity_update() and _entity_after_save()
+  // but neither gave the original values.
+  $unchanged_entity = \Drupal::entityTypeManager()
+    ->getStorage($entity->getEntityTypeId())
+    ->loadUnchanged($entity->id());
+
+  switch ($bundle) {
+    case 'disaster':
+      $fields = [
+        'name',
+        'field_glide',
+        'field_disaster_type',
+        'moderation_status',
+      ];
+      break;
+
+    case 'country':
+      $fields = [
+        'name',
+        'field_shortname',
+        'field_iso3',
+      ];
+      break;
+
+    case 'source':
+      $fields = [
+        'name',
+        'field_shortname',
+        'field_longname',
+        'field_spanish_name',
+        'field_homepage',
+        'field_disclaimer',
+        'field_organization_type',
+      ];
+      break;
+  }
+
+  foreach ($fields as $field) {
+    if ($entity->get($field)->getValue() != $unchanged_entity->get($field)->getValue()) {
+      $needs_reindex = TRUE;
+      break;
+    }
+  }
+
+  if ($needs_reindex) {
```

Comment:
Adding to the queue should be done in the `reliefweb_api_entity_after_save` to ensure the new data is in the database when the reindexing kicks in.

So in `reliefweb_api_check_reindexing` we should add a flag to the entity object:

```php
  if ($needs_reindex) {
    $entity->needs_reindex = TRUE;
  }
```

And in the `reliefweb_api_entity_after_save`:

```php
  if (!empty($entity->needs_reindex)) {
    $reindex_queue = \Drupal::state()->get('reliefweb_api.reindex_queue');
    $reindex_queue[$entity->bundle()][$entity->id()] = $entity->id();
    \Drupal::state()->set('reliefweb_api.reindex_queue', $reindex_queue);
  }
```